sage_acsv/asymptotics.py

```diff
@@ -316,6 +308,293 @@ def diagonal_asy(
+    * ``r`` -- A vector of length d of positive algebraic numbers (generally integers).
+      Defaults to the appropriate vector of all 1's if not specified.
+    * ``linear_form`` -- (Optional) A linear combination of the input
+      variables that separates the critical point solutions.
+    * ``expansion_precision`` -- A positive integer value. This is the number of terms to
+      compute in the asymptotic expansion. Defaults to 1, which only computes the leading
+      term.
+    * ``return_points`` -- If ``True``, also returns the coordinates of
+      minimal critical points. By default ``False``.
+    * ``output_format`` -- (Optional) A string or :class:`.OutputFormat` specifying
+      the way the asymptotic growth is returned. Allowed values currently are:
+      - ``"tuple"`` or ``None``, the default: the growth is returned as a list of
+        tuples of the form ``(a, n^b, pi^c, d)`` such that the `r`-diagonal of `F`
+        is the sum of ``a^n n^b pi^c d + O(a^n n^{b-1})`` over these tuples.
+      - ``"symbolic"``: the growth is returned as an expression from the symbolic
+        ring ``SR`` in the variable ``n``.
+      - ``"asymptotic"``: the growth is returned as an expression from an appropriate
+        ``AsymptoticRing`` in the variable ``n``.
+    * ``as_symbolic`` -- deprecated in favor of the equivalent
+      ``output_format="symbolic"``. Will be removed in a future release.
+    * ``whitney_strat`` -- (Optional) The user can pass in a Whitney Stratification of V(H)
+        to save computation time. The program will not check if this stratification is correct.
+        The whitney_strat should be an array of length ``d``, with the ``k``-th entry a list of
+        tuples of ideal generators representing a component of the ``k``-dimensional stratum
+
+    OUTPUT:
+
+    A representation of the asymptotic main term, either as a list of tuples,
+    or as a symbolic expression.
+
+    NOTE:
+
+    The code randomly generates a linear form, which for generic rational functions
+    separates the solutions of an intermediate polynomial system with high probability.
+    This separation step can fail, but (assuming `F` has a finite number of critical
+    points) the code can be rerun until a separating form is found.
+
+    Examples::
+
+    sage: from sage_acsv import diagonal_asy
+    sage: var('x,y')
+    (x, y)
+    sage: diagonal_asy(1/((1-(2*x+y)/3)*(1-(3*x+y)/4)), r = [17/24, 7/24], output_format = 'asymptotic')
+    12
+
+    sage: from sage_acsv import diagonal_asy
+    sage: var('x,y,z')
+    (x, y, z)
+    sage: G = (1+x)*(1-x*y^2+x^2)
+    sage: H = (1-z*(1+x^2+x*y^2))*(1-y)*(1+x^2)
+    sage: strat = [
+            [(1-z*(1+x^2+x*y^2), 1-y, 1+x^2)],
+            [(1-z*(1+x^2+x*y^2), 1-y),(1-z*(1+x^2+x*y^2), 1+x^2),(1-y,1+x^2)],
+            [(H,)]
+        ]
+    sage: diagonal_asy(G/H, r = [1,1,1], output_format = 'asymptotic', whitney_strat = strat)
+    0.8660254037844386*3^n/(sqrt(pi)*sqrt(n))
```

Comment:
fix indentation of example block

```suggestion
        sage: from sage_acsv import diagonal_asy
        sage: var('x,y')
        (x, y)
        sage: diagonal_asy(1/((1-(2*x+y)/3)*(1-(3*x+y)/4)), r = [17/24, 7/24], output_format = 'asymptotic')
        12

        sage: from sage_acsv import diagonal_asy
        sage: var('x,y,z')
        (x, y, z)
        sage: G = (1+x)*(1-x*y^2+x^2)
        sage: H = (1-z*(1+x^2+x*y^2))*(1-y)*(1+x^2)
        sage: strat = [
                [(1-z*(1+x^2+x*y^2), 1-y, 1+x^2)],
                [(1-z*(1+x^2+x*y^2), 1-y),(1-z*(1+x^2+x*y^2), 1+x^2),(1-y,1+x^2)],
                [(H,)]
            ]
        sage: diagonal_asy(G/H, r = [1,1,1], output_format = 'asymptotic', whitney_strat = strat)
        0.8660254037844386*3^n/(sqrt(pi)*sqrt(n))
```